packages/memory-inspector/src/browser/memory-widget/memory-options-widget.tsx

```diff
@@ -0,0 +1,726 @@
+                <div className={`t-mv-memory-fetch-error${this.showMemoryError ? ' show' : ' hide'}`}>
+                    {this.memoryReadError}
+                </div>
+            </>
+        );
+    }
+
+    protected activateHeaderInputField = (e: React.MouseEvent<HTMLDivElement> | React.KeyboardEvent<HTMLDivElement>): void => {
+        if (!this.isTitleEditable) {
+            const isMouseDown = !('key' in e);
+            const isActivationKey = 'key' in e && (
+                KeyCode.createKeyCode(e.nativeEvent).key?.keyCode === Key.SPACE.keyCode
+                || KeyCode.createKeyCode(e.nativeEvent).key?.keyCode === Key.ENTER.keyCode
+            );
+            if (isMouseDown || isActivationKey) {
+                if (isMouseDown) {
+                    e.currentTarget.blur();
+                }
+                this.isTitleEditable = true;
+                this.update();
+            }
+        }
+    };
+
+    protected saveHeaderInputValue = (e: React.MouseEvent<HTMLDivElement> | React.KeyboardEvent<HTMLDivElement>): void => {
+        const isMouseDown = !('key' in e);
+        const isSaveKey = 'key' in e && e.key === 'Enter';
+        const isCancelKey = 'key' in e && e.key === 'Escape';
+        e.stopPropagation();
+        if (isMouseDown || isSaveKey || isCancelKey) {
+            if (!isCancelKey && this.headerInputField) {
+                this.title.label = this.headerInputField.value;
+                this.title.caption = this.headerInputField.value;
+            }
+
+            this.isTitleEditable = false;
+            this.update();
+        }
+    };
+
+    protected assignHeaderInputRef = (element: HTMLInputElement): void => {
+        if (element) {
+            this.headerInputField = element;
+            element.focus();
+        }
+    };
+
+    protected renderToolbar(): React.ReactNode {
+        return (
+            <div className='memory-widget-toolbar'>
+                {this.memoryWidgetOptions.dynamic !== false && (
+                    <div className='memory-widget-auto-updates-container'>
+                        <div
+                            className={`fa fa-${this.doUpdateAutomatically ? 'unlock' : 'lock'}`}
+                            id={AUTO_UPDATE_TOGGLE_ID}
+                            title={this.doUpdateAutomatically ?
+                                nls.localize('theia/memory-inspector/memory/freeze', 'Freeze Memory View') :
+                                nls.localize('theia/memory-inspector/memory/unfreeze', 'Unfreeze Memory View')
+                            }
+                            onClick={this.toggleAutoUpdate}
+                            onKeyDown={this.toggleAutoUpdate}
+                            role='button'
+                            tabIndex={0}
+                        />
+                    </div>
+                )}
+                {this.renderEditableTitleField()}
+                <div className='toggle-settings-container'>
```

Comment:
It would be really great if this could also be moved into a separate method to support easier overriding also allow the user to add something to the toolbar after the settings.

Reply:
Done.